frontend/src/Files.js

```diff
@@ -80,25 +78,61 @@ class Files extends React.PureComponent {
         );
         return;
       }
-      this.setState({ loading: false });
       if (r.status === 200) {
         if (store.fetchError) {
           store.fetchError = null;
         }
-        return r.json().then((response) => {
-          this.setState({
-            files: response.files,
-            aggregates: response.aggregates,
-            total: response.total,
-            batchSize: response.batch_size,
-          });
-        });
+        return r.json().then((response) => callback(response));
       } else {
         store.fetchError = r;
+        if (errorCallback) {
+          errorCallback();
+        }
       }
     });
   };
 
+  _fetchFilesAndAggregatesAsync = (updateHistory = true) => {
+    this._fetchFiles(updateHistory);
+    this._fetchAggregates();
+  };
+
+  _fetchFiles = (updateHistory = true) => {
+    var callback = (response) => {
+      this.setState({
+        files: response.files,
+        hasNextPage: response.has_next,
+        batchSize: response.batch_size,
+      });
+      this.setState({ loadingFiles: false });
+    };
+    var errorCallback = () => {
+      this.setState({ loadingFiles: false });
+    };
+    this.setState({ loadingFiles: true });
+    this._fetch(
+      "/api/uploads/files/content",
```

Comment:
This needs a `/` at the end. Otherwise, Django returns an HTTP 301 and this follows the redirect and does a second HTTP request.

Reply:
fixed